uClibc does not have backtrace support and RTLD_NOLOAD support
So we disable them for uClibc

-Khem

Index: mozilla-central/memory/mozalloc/mozalloc.cpp
===================================================================
--- mozilla-central.orig/memory/mozalloc/mozalloc.cpp
+++ mozilla-central/memory/mozalloc/mozalloc.cpp
@@ -236,7 +236,7 @@ moz_malloc_usable_size(void *ptr)
     if (!ptr)
         return 0;
 
-#if defined(MOZ_MEMORY)
+#if defined(MOZ_MEMORY) && !defined(__UCLIBC__)
     return malloc_usable_size(ptr);
 #elif defined(XP_MACOSX)
     return malloc_size(ptr);
Index: mozilla-central/xpcom/threads/nsThread.cpp
===================================================================
--- mozilla-central.orig/xpcom/threads/nsThread.cpp
+++ mozilla-central/xpcom/threads/nsThread.cpp
@@ -52,9 +52,11 @@
 
 #if defined(XP_UNIX) && !defined(ANDROID) && !defined(DEBUG) && HAVE_UALARM \
   && defined(_GNU_SOURCE)
-# define MOZ_CANARY
 # include <unistd.h>
+#if !defined(__UCLIBC__)
+# define MOZ_CANARY
 # include <execinfo.h>
+#endif
 # include <signal.h>
 # include <fcntl.h>
 # include "nsXULAppAPI.h"
Index: mozilla-central/ipc/chromium/src/base/debug_util_posix.cc
===================================================================
--- mozilla-central.orig/ipc/chromium/src/base/debug_util_posix.cc
+++ mozilla-central/ipc/chromium/src/base/debug_util_posix.cc
@@ -11,7 +11,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
-#ifndef ANDROID
+#if !defined ANDROID && !defined __UCLIBC__
 #include <execinfo.h>
 #include <sys/sysctl.h>
 #endif
@@ -119,7 +119,7 @@ StackTrace::StackTrace() {
   const int kMaxCallers = 256;
 
   void* callers[kMaxCallers];
-#ifndef ANDROID
+#if !defined ANDROID && !defined __UCLIBC__
   int count = backtrace(callers, kMaxCallers);
 #else
   int count = 0;
@@ -138,7 +138,7 @@ StackTrace::StackTrace() {
 
 void StackTrace::PrintBacktrace() {
   fflush(stderr);
-#ifndef ANDROID
+#if !defined ANDROID && !defined __UCLIBC__
   backtrace_symbols_fd(&trace_[0], trace_.size(), STDERR_FILENO);
 #endif
 }
Index: mozilla-central/security/nss/lib/freebl/stubs.c
===================================================================
--- mozilla-central.orig/security/nss/lib/freebl/stubs.c
+++ mozilla-central/security/nss/lib/freebl/stubs.c
@@ -538,8 +538,13 @@ freebl_InitNSSUtil(void *lib)
 /*
  * fetch the library if it's loaded. For NSS it should already be loaded
  */
+#ifdef __UCLIBC__
+#define freebl_getLibrary(libName)  \
+    dlopen (libName, RTLD_LAZY)
+#else
 #define freebl_getLibrary(libName)  \
     dlopen (libName, RTLD_LAZY|RTLD_NOLOAD)
+#endif
 
 #define freebl_releaseLibrary(lib) \
     if (lib) dlclose(lib)
